src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -220,31 +220,34 @@ public ResponseEntity<StudentExam> updateWorkingTime(@PathVariable Long courseId
+     * @param examId                the exam to which the student exams belong to
+     * @param studentExamFromClient the student exam with exercises, participations and submissions
      * @return empty response with status code:
      *         200 if successful
      *         400 if student exam was in an illegal state
      */
     @PostMapping("/courses/{courseId}/exams/{examId}/student-exams/submit")
     @EnforceAtLeastStudent
-    public ResponseEntity<StudentExam> submitStudentExam(@PathVariable Long courseId, @PathVariable Long examId, @RequestBody StudentExam studentExam) {
+    public ResponseEntity<Void> submitStudentExam(@PathVariable Long courseId, @PathVariable Long examId, @RequestBody StudentExam studentExamFromClient) {
         long start = System.nanoTime();
-        log.debug("REST request to mark the studentExam as submitted : {}", studentExam.getId());
+        log.debug("REST request to mark the studentExam as submitted : {}", studentExamFromClient.getId());
 
-        User currentUser = userRepository.getUserWithGroupsAndAuthorities();
+        // 1. DB Call: read
+        User currentUser = userRepository.getUser();
         // prevent manipulation of the user object that is attached to the student exam in the request body (which is saved later on into the database as part of this request)
-        if (!Objects.equals(studentExam.getUser().getId(), currentUser.getId())) {
-            throw new AccessForbiddenException();
+        if (!Objects.equals(studentExamFromClient.getUser().getId(), currentUser.getId())) {
+            throw new AccessForbiddenException("Current user is not the user of the requested student exam");
         }
 
-        StudentExam existingStudentExam = studentExamRepository.findByIdWithExercisesElseThrow(studentExam.getId());
-        this.studentExamAccessService.checkStudentExamAccessElseThrow(courseId, examId, existingStudentExam, currentUser);
+        // 2. DB Call: read
+        StudentExam existingStudentExam = studentExamRepository.findByIdWithExercisesElseThrow(studentExamFromClient.getId());
+        checkExamConfigElseThrow(studentExamFromClient, examId, courseId);
 
-        if (Boolean.TRUE.equals(studentExam.isSubmitted()) || Boolean.TRUE.equals(existingStudentExam.isSubmitted())) {
-            log.error("Student exam with id {} for user {} is already submitted.", studentExam.getId(), currentUser.getLogin());
-            throw new ConflictException("You have already submitted.", "studentExam", "alreadySubmitted");
+        if (Boolean.TRUE.equals(studentExamFromClient.isSubmitted()) || Boolean.TRUE.equals(existingStudentExam.isSubmitted())) {
+            log.error("Student exam with id {} for user {} is already submitted.", studentExamFromClient.getId(), currentUser.getLogin());
+            // NOTE: we should not send an error message to the user here, due to overload it could happen that the call is sent multiple times
+            return ResponseEntity.ok().build();
```

Comment:
depends on whether the focus is on the action (submit exam) which is typically a POST, or on changing the object, which would be a PATCH here, because not all values are changed. I think we can stick with POST to focus on the action